src/vault_backup/backup.py

```diff
@@ -0,0 +1,81 @@
+import os
+
+import json
+import logging
+
+from lib.export_backup import (
+  backup_to_s3,
+  write_file
+)
+from lib.util import url_host_port
+from lib.vault import Vault
+
+
+def backup_secrets(
+  artsy_env,
+  vault_host,
+  vault_port,
+  kvv2_mount_point,
+  local_dir,
+  s3,
+  s3_bucket,
+  s3_prefix
+):
+  ''' export secrets '''
+  logging.info('Discovering k8s app secrets...')
+  vault_path = 'kubernetes/apps/'
+  vault_client = Vault(
+    url_host_port(vault_host, vault_port),
+    auth_method='iam',
+    kvv2_mount_point=kvv2_mount_point,
+    path=vault_path,
+  )
+  apps = vault_client.list(only_valid=False)
+  logging.debug(f'Found apps: {apps}')
+  secrets = {}
```

Comment:
It's easier to use the `defaultdict` package (so you can remove L38):

```
from collections import defaultdict
a = defaultdict(lambda: {})
```

Reply:
adopted. thanks!